### PR TITLE
suraj

### PR DESCRIPTION
**Pull request by surajxkulkarni**

### CODE REVIEW

**Comment by shanthanu9** [comment]
Next time in the title mention Cryptopals somewhere. So that we know you are submitting the assignment for Cryptopals.

**Comment by shanthanu9** [comment]
Good job writing code in C! There is small error in 2nd challenge code. Could you figure it out and rectify it? (It is a trivial error becuase of which there is an error in output)

**Comment by surajxkulkarni** [comment]
Pardon me, next time i'll mention cryptopals :p

Yeah i tried figuring out the error, but cant seem to identify it 😬. I
feel the logic is fine, i'm confused why i'm not getting the full output..

**Comment by shanthanu9** [comment]
It's a very small error:) The error is in hex2dec.

Bonus: Once you figure out the error, could you guess why the error in hex2dec did not affect your solution in the first challenge?

**Comment by surajxkulkarni** [comment]
Yeah i am confused because it isnt giving an error in the 1st challenge
The problem seems to be occurring when 'a' and '5' is passed through
hex2dec and subsequent xor... but i cant identify the error in the function
in this program...

File: Cryptopals/suraj/Set1Challenge1.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#include <string.h>
+
+char input[] = "49276d206b696c6c696e6720796f757220627261696e206c696b65206120706f69736f6e6f7573206d757368726f6f6d";
+char output[200];
+char base64_lookup_table[64] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+
+unsigned int hex2dec(char x)
+{
+  char val = 0;
```
Comment (shanthanu9) [inline Cryptopals/suraj/Set1Challenge1.c:11]: Check what the function returns when x = 'a' . Go over each line in this function carefully.

**Comment by surajxkulkarni** [comment]
Ah got it, it should be if(x>='a'). Silly mistake😅. Thats the only error
right?

**Comment by surajxkulkarni** [comment]
Thanks for pointing it out!

**Comment by shanthanu9** [comment]
Yep, that is the only error. Rectify the error and push the correct code.

**Pull request merged**